src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -1884,7 +1884,7 @@ void program::load(cldnn::BinaryInputBuffer& ib) {
 
         std::shared_ptr<cldnn::primitive> prim;
         ib >> prim;
-        if (auto data_prim = dynamic_cast<cldnn::data*>(prim.get())) {
+        if (auto data_prim = dynamic_cast<cldnn::data*>(prim.get()) && weights_memory != nullptr)  {
```

Comment:
I think this should be `OPENVINO_ASSERT(weights_memory != nullptr)`

Reply:
As [ie-tests-linux-ubuntu22-cldnn_unit_dg2/6907/consoleText](https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/ie-tests-linux-ubuntu22-cldnn_unit_dg2/6907/consoleText), there are some cases that the weights_memory is nullptr.

Reply:
move assert to where issue variable is used.

Reply:
with such change, does it resolve the issue from coverity?

Reply:
I'm not sure but it's pointing to this code at the coverity site.